### PR TITLE
Showcase conformance images options in the manifest

### PR DESCRIPTION
**Pull request by dims**
The Dockerfile has a few more options, let's surface that in the
manifest. Also use the same defaults in the manifest as the Dockerfile
itself.

Change-Id: Ib7419cf7999430db15f39ac414c80ee362fcda76

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The Dockerfile has a few more options, let's surface that in the
manifest. Also use the same defaults in the manifest as the Dockerfile
itself.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/images/conformance/conformance-e2e.yaml
Comment (stevesloka) [inline cluster/images/conformance/conformance-e2e.yaml]: The current Dockerfile used a `1` for this arg. Setting to a `y` would mean we'd pass `--nodes=25`. Is that desired for the default?

Comment (dims) [reply cluster/images/conformance/conformance-e2e.yaml]: @stevesloka yes, i'd like to try this as the default (at least for the CI jobs). sonobuoy can still use the existing defaults.

File: cluster/images/conformance/conformance-e2e.yaml
Comment (johnSchnake) [inline cluster/images/conformance/conformance-e2e.yaml]: So we deliberately dont skip any tests when running conformance tests in CI?

Sonobuoy currently defaults to skip = `'Alpha|\[(Disruptive|Feature:[^\]]+|Flaky)\]'` but I've not really known where that comes from historically. Off the top of my head I"m not sure how many (if any) fall into that category. I think some feature tests do though.

Comment (dims) [reply cluster/images/conformance/conformance-e2e.yaml]: we talked about this in https://github.com/kubernetes/test-infra/pull/9104 but i mis-remembered the outcome. will switch up to use the skip list from there:
https://github.com/heptio/sonobuoy/blob/master/pkg/client/mode.go#L42

Comment (dims) [reply cluster/images/conformance/conformance-e2e.yaml]: @timothysc @spiffxp ^^

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

File: cluster/images/conformance/conformance-e2e.yaml
Comment (spiffxp) [inline cluster/images/conformance/conformance-e2e.yaml]: ... is this skip list still necessary? I thought we had stripped out all Conformance tests that had this 

https://github.com/kubernetes/test-infra/issues/9697#issuecomment-427067895

**Comment by spiffxp** [comment]
/hold
can we strip out that skip list?  if not, what tests need to be addressed?  I want to live in a world where teh conformance images focuses on conformance and that's it, none of this skipped-but-still-labeled-as-conformance nonsense

File: cluster/images/conformance/conformance-e2e.yaml
Comment (BenTheElder) [reply cluster/images/conformance/conformance-e2e.yaml]: I don't think we can do that seeing as there are `[Serial]` tests

Comment (BenTheElder) [reply cluster/images/conformance/conformance-e2e.yaml]: also at least on kind, tentative CI results suggest that running the non-serial tests in parallel is indeed flakier (and we've been running both for some time now)

https://testgrid.k8s.io/conformance-kind#kind,%20master%20(dev)%20%5Bnon-serial%5D

**Comment by spiffxp** [comment]
I tried dry running e2e.test with a focus on Conformance, with the skip list, and without.  The number of tests didn't change.  We can inspect runs once this merges to see if there are any that I missed.

Thanks @dims for letting me patch this in.

**Pull request merged**
